### PR TITLE
[Improve][e2e] connectors e2e use separate module

### LINKED ISSUE #2607: [Improve][e2e] connectors e2e use separate module
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Currently there may be dependency conflicts when executing e2e tests in an e2e module;
The relocation of maven-shade-plugin can only solve some conflicts, for Class.forName(), relocation will cause new problems; 
so the e2e test of the connector can only be divided into different modules;

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hailin0**
## Purpose of this pull request

close #2607

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -462,16 +443,6 @@
...
-        <dependency>
-            <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter-engine</artifactId>
-        </dependency>
-
-        <dependency>
-            <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter-params</artifactId>
-        </dependency>
-
```
Comment (ashulin) [inline pom.xml]: What's the reason for the change here?

Comment (hailin0) [reply pom.xml]: reverted

**Comment by hailin0** [comment]
reference
https://maven.apache.org/plugins/maven-jar-plugin/examples/create-test-jar.html
https://maven.apache.org/plugins/maven-jar-plugin/test-jar-mojo.html#skip
https://issues.apache.org/jira/browse/MJAR-138

**Comment by hailin0** [comment]
please help to review
@ashulin @Hisoka-X @EricGao888 @CalvinKirs @TyrantLucifer @2013650523 @531651225 @laglangyue @iture123

**Review by EricJoy2048 — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]
LGTM.

**Review by Hisoka-X — APPROVED** [review]
LGTM
